File: app-android/src/main/java/com/ustadmobile/port/android/view/SelectableSpinner.java
package com.ustadmobile.port.android.view;

import android.content.Context;
import android.util.AttributeSet;

public class SelectableSpinner extends android.support.v7.widget.AppCompatSpinner {

    OnItemSelectedListener listener;

    public SelectableSpinner(Context context, AttributeSet attrs)
    {
        super(context, attrs);
    }

    @Override
    public void setSelection(int position)
    {
        super.setSelection(position);

        if (position == getSelectedItemPosition())
        {
            listener.onItemSelected(null, null, position, 0);
        }
    }

    public void setOnItemSelectedListener(OnItemSelectedListener listener)
    {
        this.listener = listener;
    }
}

File: app-android/src/main/kotlin/com/ustadmobile/port/android/view/SelectableSpinner.kt
package com.ustadmobile.port.android.view

import android.content.Context
import android.util.AttributeSet
import android.widget.AdapterView
import androidx.appcompat.widget.AppCompatSpinner

class SelectableSpinner(context: Context, attrs: AttributeSet) : AppCompatSpinner(context, attrs) {

    lateinit var listener: OnItemSelectedListener

    override fun setSelection(position: Int) {
        super.setSelection(position)

        if (position == getSelectedItemPosition()) {
            listener.onItemSelected(null, null, position, 0)
        }
    }

    override fun setOnItemSelectedListener(listener: OnItemSelectedListener?) {
        super.setOnItemSelectedListener(listener)
        this.listener = listener!!
    }
}
